promql/engine.go

```diff
@@ -1064,6 +1068,10 @@ type evaluator struct {
 	samplesStats             *stats.QuerySamples
 	noStepSubqueryIntervalFn func(rangeMillis int64) int64
 	enableDelayedNameRemoval bool
+	querier                  storage.Querier
+
+	// selectHints is the top-most SelectHints in the expression tree.
```

Comment:
In the code it is not topmost; it is whatever VectorSelector was executed last.
This can't be correct.

Reply:
Agree your interpretation is more correct. I've pushed a revision of this doc comment.

Reply:
Dropped this field.